compiler-cli/src/run.rs

```diff
@@ -174,9 +181,14 @@ fn run_erlang_command(
 
     // gleam modules are separated by `/`. Erlang modules are separated by `@`.
     let module = module.replace('/', "@");
-
+    // Decide what to evaluate
+    let erl_eval = if use_entrypoint {
+        format!("{package}@@main:run({module})")
+    } else {
+        format!("{module}:main(), init:stop()")
+    };
```

Comment:
The entrypoint must always be used! Otherwise the virtual machine does not get booted correctly and the IO systems are not configured as Gleam code expects. Running it like this will cause many or most projects to fail when run.

Reply:
No, the entrypoint must be run otherwise the application does not get booted.

The entrypoint need to be made to boot the correct OTP application for the module being run https://github.com/gleam-lang/gleam/issues/4771#issuecomment-3061859908

Reply:
That's right!

> P.S. Do you have any projects that will fail with this setting so I could use them as an example to compare/experiment with?

If you make a project and define an application start module for it then it will fail.